File: 3. Arrays/1. Easy/Python/4. Remove duplicate elements from sorted array.py
# cook your dish here
#Remove duplicate elements from sorted array.
import sys

def removeDuplicate(arr,n):
    i = 0
    for j in range(1,n):
        if(arr[j]!=arr[i]):
            arr[i+1] = arr[j]
            i = i+1
    return i+1


n = int(input())
arr = list(map(int,input().strip().split()))
size = removeDuplicate(arr,n)
for i in range(0,size):
    print(arr[i],end=" ")

File: 3. Arrays/1. Easy/C++/4. Remove duplicate elements from sorted array.cpp
/*
Remove duplicate elements from sorted array.
*/

#include <bits/stdc++.h>
using namespace std;


int remove_duplicate(int arr[],int n){
    // code here
    int i = 0;
    for(int j=1;j<n;j++){
        if(arr[j]!=arr[i]){
            arr[i+1] = arr[j];
            i++;
        }
    }
    return i+1;
}

int main() {
	// your code goes here
    int n;
    cin >> n;
    int arr[n];
    for(int i=0;i<n;i++){
        cin >> arr[i];
    }
    int size = remove_duplicate(arr,n);
    for(int i=0;i<size;i++){
        cout << arr[i] << " ";
    }
}
